src/commands/containers/files/downloadContainerFile.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { IActionContext, UserCancelledError } from 'vscode-azureextensionui';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../localize';
+import { FileTreeItem } from '../../../tree/containers/files/FileTreeItem';
+import { multiSelectNodes } from '../../../utils/multiSelectNodes';
+
+async function fileExists(file: vscode.Uri): Promise<boolean> {
+    try {
+        // NOTE: The expectation is that stat() throws when the file does not exist.
+        //       Filed https://github.com/microsoft/vscode/issues/112107 to provide
+        //       a better mechanism than trapping exceptions.
+        await vscode.workspace.fs.stat(file);
+
+        return true;
+    } catch {
+        return false;
+    }
+}
+
+export async function downloadContainerFile(context: IActionContext, node?: FileTreeItem, nodes?: FileTreeItem[]): Promise<void> {
+    nodes = await multiSelectNodes(
+        { ...context, noItemFoundErrorMessage: localize('vscode-docker.commands.containers.files.downloadContainerFile.noFiles', 'No files are available to download.') },
+        ext.containersTree,
+        'containerFile',
+        node,
+        nodes
+    );
+
+    const localFolderUris = await vscode.window.showOpenDialog(
+        {
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+            openLabel: localize('vscode-docker.commands.containers.files.downloadContainerFile.openLabel', 'Select'),
+            title: localize('vscode-docker.commands.containers.files.downloadContainerFile.openTitle', 'Select folder for download')
+        });
+
+    const localFolderUri = localFolderUris[0];
```

Comment:
According to docs `showOpenDialog` can return `undefined`? Should we handle that?

Is it a possibility that it will return zero-length array?

Reply:
Good catch.  I'll accommodate users canceling the dialog.  A zero-length array *shouldn't* happen, but I'll lump that into the cancellation case as well.